qiskit_bot/community.py

```diff
@@ -17,16 +17,15 @@
 
 def add_community_label(pr_data, repo):
     """Add community label to PR when author not associated with core team"""
-    # check repo is monitored by community review team
-    if repo.repo_config.get('uses_community_label'):
-        # check if PR was authored by soemone outside core repo team
-        if (pr_data['pull_request']['author_association'] != 'MEMBER'
-            ) and (pr_data['pull_request']['user']['type']
-                   not in EXCLUDED_USER_TYPES):
-            # fetch label data
-            labels = pr_data['pull_request']['labels']
-            label_names = [label['name'] for label in labels]
-            # tag PR with 'community PR's
-            if "Community PR" not in label_names:
-                pr = repo.gh_repo.get_pull(pr_data['pull_request']['number'])
-                pr.add_to_labels("Community PR")
+    if any((
+        not repo.repo_config.get("uses_community_label", False),
+        pr_data["user"]["type"] in EXCLUDED_USER_TYPES,
+        "Community PR" in [label["name"] for label in pr_data["labels"]],
+    )):
+        return
+    # We need to use the bot's API key rather than public data to know if the
+    # user is a private member of the organisation.  PyGitHub doesn't expose
+    # the 'author_association' attribute as part of the typed interface.
+    pr = repo.gh_repo.get_pull(pr_data["number"])
+    if pr.raw_data["author_association"] not in ("MEMBER", "OWNER"):
```

Comment:
@javabster: probably best answered by you and the rest of the community team.

I didn't add it in the list because I thought "external contributor" is more likely to be a community member (e.g. someone like Edwin), but I don't actually know if we use the role, and how if so.

Reply:
Oh, well I  assumed this was the collaborator role is like when you give permissions on a repo outside the organization. This is the role I see for my co-maintainers on repos on my personal github. That's why I wasn't sure it would apply here but we can always add it later if we so desire. Let's merge this and get the fix deployed

Reply:
Yeah, I don't actually know how we use it (if) at all.

Reply:
sorry for the slow reply, this got buried in my notifications. I also don't really know the scope of collaborator and as none of us are clear on it i would say err on the side of caution and not add it